pyquil/quil.py

```diff
@@ -949,10 +949,14 @@ def merge_programs(prog_list):
             seen[name] = [definition]
     new_definitions = [gate for key in seen.keys() for gate in reversed(seen[key])]
 
-    p = sum([Program(prog).instructions for prog in prog_list], Program())  # Combine programs without gate definitions
+    # Combine programs without gate definitions; avoid call to _synthesize by using _instructions
+    p = sum([Program(prog)._instructions for prog in prog_list], Program())
```

Comment:
The `Program` in `Program(prog)._instructions` seems redundant. I think `prog._instructions` ought to suffice. And actually I think `p = Program(*[prog._instructions for prog in prog_list])` will achieve the same thing.